torch/csrc/Runtime.cpp

```diff
@@ -0,0 +1,65 @@
+#include <torch/csrc/python_headers.h>
+#include <fmt/format.h>
+#include <pybind11/chrono.h>
+#include <torch/csrc/utils/pybind.h>
+#include <torch/csrc/Runtime.h>
+#include <c10/core/DeviceType.h>
+
+namespace torch {
+
+std::map<DeviceType, PyObject*> runtime_dict;
+
+template <typename T>
+using shared_ptr_class_ = py::class_<T, std::shared_ptr<T>>;
+
+static py::module get_runtime_module(DeviceType dev_type) {
+  TORCH_CHECK(runtime_dict.find(dev_type) != runtime_dict.end(), "Runtime '", dev_type, "' not exists.");
+  return py::handle(runtime_dict[dev_type]).cast<py::module>();
+}
+
+void register_device_runtime(const DeviceType dev_type, PyObject* module) {
+  auto new_module = py::handle(module).cast<py::module>();
+  auto _torch = THPObjectPtr(PyImport_ImportModule("torch"));
+  if (!_torch) {
+    throw python_error();
+  }
+  TORCH_CHECK(runtime_dict.find(dev_type) == runtime_dict.end(), "Runtime '", dev_type, "' already registered. ");
```

Comment:
should override existing one?

Reply:
Or, any risk to override.

Reply:
This is to protect registering the runtime module from mistake.

Is there any use case for overriding a device runtime module?